runtime/nvqir/qpp/QppCircuitSimulator.cpp

```diff
@@ -25,62 +25,40 @@ class QppCircuitSimulator : public nvqir::CircuitSimulatorBase<double> {
   StateType state;
 
   /// Convert from little endian to big endian.
-  std::size_t bigEndian(const int n_qubits, const int bit) {
-    return n_qubits - bit - 1;
+  std::size_t bigEndian(const int bit) {
+    return std::log2(stateDimension) - bit - 1;
```

Comment:
While working on #668, I encountered the fact that `std::log2(0)` produces different results on Mac vs Linux. It would be best to specifically check for that case and return a defined value in that case (even if we don't expect it to occur right now) as it may help save debugging in the future.

Reply:
As `std::log2` comes from C, `std::log2(0)` should set a floating-point exception flag and a `errno` flag. They are both annoying to check, and, in this case, it wound't make sense to return any value if the dimension is `0` (unless we add the notion of invalid qubit index). I will add an assertion for the final version of this PR.

Note: In reality, I didn't want to retrieve the number of qubits in this way. I thought that `nQubitsAllocated` should do the trick, but it doesn't (this variable has a misleading name, as it doesn't represent the number of allocated qubits, but the number of "live"/"in use" qubits).